### PR TITLE
add RHACS deployment, image check and scan tasks

### PR DESCRIPTION
**Pull request by dlbewley**
# Changes

Adds tasks `rhacs-deployment-check`, `rhacs-image-check`, and `rhacs-image-scan` using `roxctl` from Red Hat Advanced Cluster Security.

Developed with input from tasks stackrox-image-check and stackrox-image-scan with these differences

* `roxctl` image rather than curling for binary
* command args rather than script
* tasks are versioned to match roxctl image
* updated examples help illustrate forced image scan

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Follows the [authoring recommendations][authoring]
- [x] Includes [docs][docs] (if user facing)
- [ ] Includes [tests][tests] (for new tasks or changed functionality)
  - **No**. _Requires external service integration._
  - See the [end-to-end testing documentation][e2e] for guidance and CI details.
- [x] Meets the [Tekton contributor standards][contributor] (including functionality, content, code)
- [x] Commit messages follow [commit message best practices][commit]
- [x] Has a kind label. You can add one by adding a comment on this PR that
  contains `/kind <type>`. Valid types are bug, cleanup, design, documentation,
  feature, flake, misc, question, tep
- [x] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [x] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [x] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [x] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [x] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [x] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

/kind feature

---

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]: https://github.com/tektoncd/catalog/issues/413
[authoring]: https://github.com/tektoncd/catalog/blob/main/recommendations.md
[docs] https://github.com/tektoncd/community/blob/master/standards.md#docs
[tests] https://github.com/tektoncd/community/blob/master/standards.md#tests
[e2e] https://github.com/tektoncd/catalog/blob/main/CONTRIBUTING.md#end-to-end-testing
[contributor] https://github.com/tektoncd/community/blob/main/standards.md
[commit] https://github.com/tektoncd/community/blob/master/standards.md#commit-messages

### CODE REVIEW

File: task/rhacs-deployment-check/3.71/rhacs-deployment-check.yaml
Comment (janisz) [inline task/rhacs-deployment-check/3.71/rhacs-deployment-check.yaml:81]: I'm afraid it's not working like this. This flag takes no args.

File: task/rhacs-image-check/3.71/rhacs-image-check.yaml
Comment (janisz) [inline task/rhacs-image-check/3.71/rhacs-image-check.yaml:80]: ditto

**Review by vdemeester — APPROVED** [review]
/cc @vinamra28

File: task/rhacs-image-check/3.71/README.md
Comment (vinamra28) [inline task/rhacs-image-check/3.71/README.md:7]: @dlbewley, the following Task already exists in the catalog with the name [stackrox-image-check](https://github.com/tektoncd/catalog/tree/main/task/stackrox-image-check/0.1) (I guess). If so, can we please bump the version of the existing task if there are any changes?

File: task/rhacs-image-scan/3.71/README.md
Comment (vinamra28) [inline task/rhacs-image-scan/3.71/README.md:5]: @dlbewley, the following Task already exists in the catalog with the name [stackrox-image-scan](https://github.com/tektoncd/catalog/tree/main/task/stackrox-image-scan/0.1) (I guess). If so, can we please bump the version of the existing task if there are any changes?

File: task/rhacs-deployment-check/3.71/rhacs-deployment-check.yaml
Comment (vinamra28) [inline task/rhacs-deployment-check/3.71/rhacs-deployment-check.yaml:54]: can we parameterise the image with `quay.io/stackrox-io/roxctl:3.71.0` as the default value. This will enable users to use the image of their choice if they want to

File: task/rhacs-deployment-check/3.71/rhacs-deployment-check.yaml
Comment (vinamra28) [inline task/rhacs-deployment-check/3.71/rhacs-deployment-check.yaml:80]: as far as I remember, environment variables are references using `${var}`
```suggestion
        - --endpoint=${ROX_CENTRAL_ENDPOINT}
```

@vdemeester @chmouel please correct me if I am wrong 😅

File: task/rhacs-deployment-check/3.71/rhacs-deployment-check.yaml
Comment (vinamra28) [inline task/rhacs-deployment-check/3.71/rhacs-deployment-check.yaml:34]: can this enable user to provide a `deployment.yaml` which is present at a place which can't be cloned? If so, can we assume that workspace can be made optional in that case?

File: task/rhacs-deployment-check/3.71/README.md
Comment (vinamra28) [inline task/rhacs-deployment-check/3.71/README.md:14]: instead of using raw.githubusercontent URL can we use the endpoint which tekton hub would be generating once this Task is merged? Format of the endpoint that Hub is going to generate would be
```
https://api.hub.tekton.dev/v1/resource/tekton/task/{TaskName}/{TaskVersion}/raw
```
so in your case the endpoint generated would be
```
https://api.hub.tekton.dev/v1/resource/tekton/task/rhacs-deployment-check/3.71/raw
```

**Review by vinamra28 — CHANGES_REQUESTED** [review]

File: task/rhacs-deployment-check/3.71/rhacs-deployment-check.yaml
Comment (dlbewley) [reply task/rhacs-deployment-check/3.71/rhacs-deployment-check.yaml:80]: In testing, in this context, I found that parens were required for interpolation to occur as opposed to braces. Eg. changing to braces results in

> "transport: authentication handshake failed: dial tcp: address ${ROX_CENTRAL_ENDPOINT}: missing port in address"

File: task/rhacs-deployment-check/3.71/rhacs-deployment-check.yaml
Comment (dlbewley) [reply task/rhacs-deployment-check/3.71/rhacs-deployment-check.yaml]: good idea. implemented. it did result in a warning from Catlin

File: task/rhacs-deployment-check/3.71/rhacs-deployment-check.yaml
Comment (dlbewley) [reply task/rhacs-deployment-check/3.71/rhacs-deployment-check.yaml:37]: the working dir is set to the workspace path so a relative filename may be used.
i can not think of where the file may come from if not from a git clone in a workspace.

File: task/rhacs-deployment-check/3.71/README.md
Comment (dlbewley) [reply task/rhacs-deployment-check/3.71/README.md]: thanks for that. done.

File: task/rhacs-image-check/3.71/README.md
Comment (dlbewley) [reply task/rhacs-image-check/3.71/README.md:7]: @vinamra28 these tasks are developed specifically to target RHACS installations rather than the upstream solution. i am amenable to contributing enhancements to those tasks as appropriate in a subsequent PR.

File: task/rhacs-image-scan/3.71/README.md
Comment (dlbewley) [reply task/rhacs-image-scan/3.71/README.md]: @vinamra28 these tasks are developed specifically to target RHACS installations rather than the upstream solution. i am amenable to contributing enhancements to those tasks as appropriate in a subsequent PR.

File: task/rhacs-deployment-check/3.71/rhacs-deployment-check.yaml
Comment (vinamra28) [reply task/rhacs-deployment-check/3.71/rhacs-deployment-check.yaml:37]: > i can not think of where the file may come from if not from a git clone in a workspace.

one such case can be - a file is present in gcs or s3 or any hosted bucket. Hope

File: task/rhacs-image-scan/3.71/README.md
Comment (vinamra28) [reply task/rhacs-image-scan/3.71/README.md]: @dlbewley, will this task not work if StackRox is installed instead of RHACS?

File: task/rhacs-deployment-check/3.71/rhacs-deployment-check.yaml
Comment (dlbewley) [reply task/rhacs-deployment-check/3.71/rhacs-deployment-check.yaml:37]: That does make sense but the tool (roxctl entrypoint) can only operate on POSIX file paths, which puts object stores out of reach of the task independently.

File: task/rhacs-deployment-check/3.71/rhacs-deployment-check.yaml
Comment (dlbewley) [reply task/rhacs-deployment-check/3.71/rhacs-deployment-check.yaml]: I think this is resolved.

File: task/rhacs-deployment-check/3.71/rhacs-deployment-check.yaml
Comment (dlbewley) [reply task/rhacs-deployment-check/3.71/rhacs-deployment-check.yaml:80]: I think this is resolved.

File: task/rhacs-image-scan/3.71/README.md
Comment (dlbewley) [reply task/rhacs-image-scan/3.71/README.md]: incorrect. it is not incompatible.

File: task/rhacs-deployment-check/3.71/rhacs-deployment-check.yaml
Comment (dlbewley) [reply task/rhacs-deployment-check/3.71/rhacs-deployment-check.yaml:37]: I think this is resolved.

**Comment by dlbewley** [comment]
Hi @vinamra28 Is there anything left which blocks this PR?

**Comment by dlbewley** [comment]
Hi @chmouel could I trouble you for a review?

/unassign @kimsterv

**Pull request merged**
